packages/cinterop/build.gradle.kts

```diff
@@ -277,74 +310,102 @@ val capiAndroidRelease by tasks.registering {
 }
 // Building Mach-O universal binary with 2 architectures: [x86_64] [arm64] (Apple M1) for macOS
 val capiMacosUniversal by tasks.registering {
-    build_C_API_Macos_Universal(releaseBuild = false) // TODO https://github.com/realm/realm-kotlin/issues/142 switch to Release build (with assertion enabled) once we reach Beta stability?
+    build_C_API_Macos_Universal(releaseBuild = isReleaseBuild)
 }
 // Building Mach-O universal binary with 2 architectures: [x86_64] [arm64] (Apple M1) for iphone simulator
 val capiSimulatorUniversal by tasks.registering {
-    build_C_API_Simulator_Universal(releaseBuild = false)
+    build_C_API_Simulator_Universal(releaseBuild = isReleaseBuild)
 }
 // Building for ios device (arm64 only)
 val capiIosArm64 by tasks.registering {
-    build_C_API_iOS_Arm64(releaseBuild = false)
+    build_C_API_iOS_Arm64(releaseBuild = isReleaseBuild)
 }
 
 fun Task.build_C_API_Android(releaseBuild: Boolean = false) {
     val buildType = if (releaseBuild) "Release" else "Debug"
     doLast {
         // x86_64
         exec {
-            workingDir(project.file("$relativeCorePath"))
-            commandLine("tools/cross_compile.sh", "-t", buildType, "-a", "x86_64", "-o", "android", "-f", "-DREALM_ENABLE_SYNC=0 -DREALM_NO_TESTS=ON")
+            workingDir(project.file("$absoluteCorePath"))
```

Comment:
Are we even using these tasks for anything? Shouldn't the CMake file handle all of this?

Reply:
These tasks are using CMake with various toolchains, this also gives us the granularity to control some flags (like `-UseModernBuildSystem=NO` more easily)

Reply:
I don't think these tasks are triggered anymore after building through the monorepo Cmake-infrastructure, so you would need to incorporate stuff into the `CMakeLists.txt`. At least I removed them in the `frozen-architecture`-branch without causing issues.